src/lib/components/TableOfContents/index.js

```diff
@@ -98,11 +98,14 @@ class TableOfContents extends BaseStateElement {
     for (const heading of headings) {
       const href = `#${heading.target.getAttribute('id')}`;
       const link = links.find((l) => l.getAttribute('href') === href);
```

Comment:
Unrelated but is it worth converting `links` into an object mapping from href to link, so we can remove this find?

Reply:
Done, well, I used a Map instead of just making an object as it's easier to iterate through if needed, but it saves the need to use the `find` method.